kolibri/plugins/management/assets/src/actions.js

```diff
@@ -25,19 +25,30 @@ const samePageCheckGenerator = require('kolibri.coreVue.vuex.actions').samePageC
  * the API to state in the Vuex store
  */
 
-function _userState(data) {
-  // assume just one role for now
+function _stateUser(apiUserData) {
+  // handle role representation
   let kind = UserKinds.LEARNER;
-  if (data.roles.length && data.roles[0].kind === 'admin') {
-    kind = UserKinds.ADMIN;
-  }
+
+  // makes kind the highest ranking role
+  apiUserData.roles.forEach(role => {
+    // using a switch statement. Checks all in order of heirarchy
+    switch(role.kind){
+      // sets role to admin 
+      case UserKinds.ADMIN || UserKinds.SUPERUSER:
+        kind = UserKinds.ADMIN;
+        break;
+      case UserKinds.COACH:
+        if(kind != UserKinds.ADMIN) kind = UserKinds.COACH;
+        break;
+    }
+  });
+
   return {
-    id: data.id,
-    facility_id: data.facility,
-    username: data.username,
-    full_name: data.full_name,
-    roles: data.roles,
-    kind, // unused for now
+    id: apiUserData.id,
+    facility: apiUserData.facility,
```

Comment:
`facility_id` was renamed to `facility`. If it is in fact an ID we should probably leave the variable name? Otherwise, let's give it something more descriptive (e.g. `facility_name` if that's what it is)